### PR TITLE
Improve unit test coverage for rustjail/container.rs 

### PR DESCRIPTION
**Pull request by devimc**

### CODE REVIEW

File: src/agent/rustjail/src/cgroups/mock.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+use libc::{self, pid_t};
+use oci::LinuxResources;
+use std::collections::HashMap;
+use std::string::String;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Manager {
+    pub paths: HashMap<String, String>,
+    pub mounts: HashMap<String, String>,
+    // pub rels: HashMap<String, String>,
```
Comment (jodh-intel) [inline src/agent/rustjail/src/cgroups/mock.rs:21]: Nit: commented out code.

File: src/agent/rustjail/src/container.rs [hunk 2]
```diff
@@ -694,21 +697,33 @@ impl BaseContainer for LinuxContainer {
...
     fn state(&self) -> Result<State> {
-        Err(anyhow!("not suppoerted"))
+        Err(anyhow!("not supported"))
     }
 
     fn oci_state(&self) -> Result<OCIState> {
-        let oci = self.config.spec.as_ref().unwrap();
+        let oci = match self.config.spec.as_ref() {
+            Some(s) => s,
+            None => return Err(anyhow!("Unable to get oci state: spec not found")),
```
Comment (jodh-intel) [inline src/agent/rustjail/src/container.rs:706]: Nit: s/oci/OCI/

File: src/agent/rustjail/src/container.rs [hunk 3]
```diff
@@ -1672,4 +1696,295 @@ mod tests {
...
+            .and_then(|p: tempfile::TempDir| {
+                // Create a new container and try to pause it
+                LinuxContainer::new(
+                    "some_id",
+                    &p.path().join("rootfs").to_str().unwrap(),
+                    create_dummy_opts(),
+                    &slog_scope::logger(),
+                )
+                .and_then(op)
+            })
```
Comment (jodh-intel) [inline src/agent/rustjail/src/container.rs:1797]: Nice! I want this on a T-shirt! :smile:

**Review by jodh-intel — APPROVED** [review]
Very nice - thanks @devimc !

**Review by GabyCT — APPROVED** [review]

**Pull request merged**